baseapp/baseapp_payments/utils.py

```diff
@@ -1,138 +1,194 @@
+            logger.exception(e)
+            return JsonResponse({"error": "Error"}, status=500)
+
+    @staticmethod
+    def customer_deleted(event):
+        customer_data = event["data"]["object"]
+        try:
+            Customer.objects.get(remote_customer_id=customer_data["id"]).delete()
+            return JsonResponse({"status": "success"}, status=200)
+        except Exception:
+            return JsonResponse({"status": "success"}, status=200)
+
+    @staticmethod
+    def subscription_created(event):
+        subscription_data = event["data"]["object"]
+        try:
+            Subscription.objects.create(
+                remote_customer_id=subscription_data["customer"],
+                remote_subscription_id=subscription_data["id"],
+            )
+            return JsonResponse({"status": "success"}, status=200)
+        except IntegrityError:
+            return JsonResponse({"status": "success"}, status=200)
+        except Exception as e:
+            logger.exception(e)
+            return JsonResponse({"error": "Error"}, status=500)
+
+    @staticmethod
+    def subscription_deleted(event):
+        subscription_data = event["data"]["object"]
+        try:
+            Subscription.objects.get(remote_subscription_id=subscription_data["id"]).delete()
+            return JsonResponse({"status": "success"}, status=200)
+        except ObjectDoesNotExist:
+            return JsonResponse({"status": "success"}, status=200)
+        except Exception as e:
+            logger.exception(e)
+            return JsonResponse({"error": "Error"}, status=500)
+
+    def webhook_handler(self, request, secret):
+        payload = request.body
+        sig_header = request.META["HTTP_STRIPE_SIGNATURE"]
+        endpoint_secret = secret
+
+        try:
+            event = stripe.Webhook.construct_event(payload, sig_header, endpoint_secret)
+        except ValueError as e:
+            logger.exception(e)
+            return JsonResponse({"error": "Error"}, status=400)
+        except stripe.error.SignatureVerificationError as e:
+            logger.exception(e)
+            return JsonResponse({"error": "Error"}, status=400)
+
+        event_type = event["type"]
+        handler = self.EVENT_HANDLERS.get(event_type)
+
+        if handler:
+            return handler(event)
+        else:
+            # if event is not handled, return 200 so stripe doesn't keep resending
+            return JsonResponse({"status": "success"}, status=200)
+
+
+class StripeService:
+    def __init__(
+        self,
+        api_key=settings.STRIPE_SECRET_KEY,
+    ):
+        stripe.api_key = api_key
+
+    def create_customer(self, email=None):
+        try:
+            if email is None:
+                return stripe.Customer.create()
+            return stripe.Customer.create(email=email)
+        except Exception as e:
+            logger.exception(e)
+            raise ValidationError("Error creating customer in Stripe")
```

Comment:
I'm not 100% sure if `ValidationError` is appropriate here (ditto for the other methods), since we're not actually validating anything.